packages/eas-cli/src/utils/profiles.ts

```diff
@@ -37,6 +42,11 @@ export async function getProfilesAsync<T extends ProfileType>({
       type,
       profileName,
     });
+
+    if (isBuildProfile(profile, type)) {
+      await setNodeVersionFromFileAsync(projectDir, profile);
+    }
```

Comment:
Maybe it's better to move this logic to https://github.com/expo/eas-cli/blob/fd99365655872dc6d7ce33e762c9bd3a99e0696a/packages/eas-cli/src/utils/profiles.ts#L61-L71

We already have some conditional logic only for building profiles there

(not a strong opinion just a suggestion)